test/unit/gateways/d_local_test.rb

```diff
@@ -62,6 +62,18 @@ def test_successful_authorize_without_address
     assert_equal 'D-15104-be03e883-3e6b-497d-840e-54c8b6209bc3', response.authorization
   end
 
+  def test_passing_billing_address
```

Comment:
😎
What do you think about another test where street or number isn't present in the address field? I don't know how likely that'd actually be but might be worth just exercising that bit of code too